File: backend/parking-domain/src/main/java/com/chatchatabc/parking/domain/service/VehicleService.java
package com.chatchatabc.parking.domain.service;

import com.chatchatabc.parking.domain.model.Vehicle;

public interface VehicleService {
    /**
     * Register vehicle
     *
     * @param userId      the user id
     * @param name        the name of the vehicle
     * @param plateNumber the plate number of the vehicle
     * @param type        the type of the vehicle
     * @return the vehicle
     */
    Vehicle registerVehicle(String userId, String name, String plateNumber, int type);

    /**
     * Update vehicle
     *
     * @param userId      the user id
     * @param vehicleId   the vehicle id
     * @param name        the name of the vehicle
     * @param plateNumber the plate number of the vehicle
     * @param type        the type of the vehicle
     * @return the vehicle
     */
    Vehicle updateVehicle(String userId, String vehicleId, String name, String plateNumber, int type);
}

File: backend/src/main/kotlin/com/chatchatabc/parking/domain/service/VehicleService.kt
package com.chatchatabc.parking.domain.service

import com.chatchatabc.parking.domain.model.Vehicle
import org.springframework.stereotype.Service

@Service
interface VehicleService {
    /**
     * Register a vehicle to a user
     */
    fun register(userId: String, vehicle: Vehicle): Vehicle

    /**
     * Update a vehicle's information
     */
    fun update(userId: String, vehicleId: String, newVehicleInfo: Vehicle): Vehicle
}